site/docs/tasks/control-plane-security.md

```diff
@@ -12,7 +12,83 @@ The `NetworkPolicy` is always created. However, your Kubernetes cluster must hav
+
+
+#### Adding TLS Secrets
+
+The manager/connectors certificates are kept in Kubernetes secret:
+
+For each component copy its certificate into a file names tls.crt. Copy the certificate key into a file named tls.key.
+
+Use kubectl with the tls secret type to create the secrets.
+
+```bash
+kubectl -n fybrik-system create secret tls tls-manager-certs \
+  --cert=tls.crt \
+  --key=tls.key
+```
+
+[Cert-manager](https://cert-manager.io/) can also be used to generate the secret above using its [`Certificate`](https://cert-manager.io/docs/concepts/certificate/) resource. For example, the following is an example of a `Certificate` resource for the manager where a tls type secret named `tls-manager-certs` is created by the cert-manager. The `issuerRef` field points to a cert-manager resource names [`Issuer`](https://cert-manager.io/docs/configuration/ca/) that holds the information about the CA that signs the certificate.
+
+```bash
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: tls-manager-cert
+  namespace: fybrik-system
+spec:
+  dnsNames:
+  - manager
+  issuerRef:
+    kind: Issuer
+    name: ca-issuer
+  secretName: tls-manager-certs
+```
+
+#### Using a Private CA Signed Certificate
+
+If you are using a private CA, Fybrik requires a copy of the CA certificate which is used by connector/manager to validate the connection to the manager/connectors.
+
+For each component copy the CA certificate into a file named cacerts.pem and use kubectl to create the tls-ca secret in the fybrik-system namespace.
+
+```bash
+kubectl -n fybrik-system create secret generic tls-ca \
+  --from-file=cacerts.pem=./cacerts.pem
+```
+
+#### Update Values.yaml file
+
+To use TLS the infomation about the secrets above should be inserted to the fields in [values.yaml](https://github.com/fybrik/fybrik/blob/master/charts/fybrik/values.yaml) file upon Fybrik deployment using helm.
+
+Here is an exmaple of the tls related fields in the manager that are filled based on the secrets created above:
+
+```bash
+# Manager component
+manager:
+  tls:
+    certs:
+      # Name of kubernetes secret that holds the manager certificate.
+      # The secret should be of `kubernetes.io/tls` type.
+      certSecretName: "tls-manager-certs"
+      # Name of kubernetes tls secret namespace that holds the manager certificate.
+      # The secret should be of `kubernetes.io/tls` type.
+      certSecretNamespace: "fybrik-system"
+      # Name of kubernetes secret that holds the certificate authority (CA) certificates
+      # which are used by the manager to validate the connection to the connectors.
+      cacertSecretName: "tls-ca"
+      # Name of kubernetes secret namespace that holds the certificate authority (CA)
+      # certificates which are used by the manager to validate the connection to the connectors.
+      cacertSecretNamespace: "fybrik-system"
+```
```

Comment:
where is the useTLS and useMTLS flags ?
maybe better to provide a server (one of teh conectors ) as an example